### PR TITLE
OSDOCS-2855 Installing on IBM Cloud IPI

### PR DESCRIPTION
**Pull request by mjpytlak**
This is a continuation of https://github.com/openshift/openshift-docs/pull/39767.

CP to 4.10

https://issues.redhat.com/browse/OSDOCS-2855

Preview

[Preparing to install on IBM Cloud](https://deploy-preview-42293--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/preparing-to-install-on-ibm-cloud.html)
[Configuring an IBM Cloud account](https://deploy-preview-42293--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-account.html)
[Configuring IAM for IBM Cloud](https://deploy-preview-42293--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.html)
[Installing a cluster on IBM Cloud with customizations](https://deploy-preview-42293--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-customizations.html)
[Installing a cluster on IBM Cloud with network customizations](https://deploy-preview-42293--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/installing-ibm-cloud-network-customizations.html)
[Uninstalling a cluster on IBM Cloud](https://deploy-preview-42293--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_public/uninstalling-cluster-ibm-cloud.html)

### CODE REVIEW

**Comment by mjpytlak** [comment]
Hi @pamoedom I am taking over the IBM Cloud doc with Cody's departure. My latest commit includes a few late breaking edits from the team. Looking for one final QE ack. This is fairly time sensitive, as doc freeze is this Friday, 2/25. Thanks in advance.

**Comment by pamoedom** [comment]
Hi @mjpytlak , thanks for taking over, one question, is the generated preview ok? I see the IBMCloud section referring IPI BM stuff which makes no sense, example:
Screenshot_2022-02-23_21-50-53
Screenshot_2022-02-23_21-53-15

**Comment by mjpytlak** [comment]
Yes. That through me off, as well, the first time I previewed the build. Per Cody during his handover.

"We need to retitle the current doc set we have on installing bare metal IPI clusters to IBM Cloud . Telco owns this, and I have a PR up to get approval from that team. John Wilkins, the tech writer responsible, has ACK’d via Slack. Looking for PM ACK in the PR https://github.com/openshift/openshift-docs/pull/39716"

I will follow up with the Teleco writer.

File: installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+include::modules/common-attributes.adoc[]
+:context: configuring-iam-ibm-cloud
+
+toc::[]
+
+In environments where the cloud identity and access management (IAM) APIs are not reachable, you must put the Cloud Credential Operator (CCO) into manual mode before you install the cluster.
+
+include::modules/alternatives-to-storing-admin-secrets-in-kube-system.adoc[leveloffset=+1]
+
+.Additional resources
```
Comment (ahardin-rh) [inline installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc:12]: [role="_additional-resources"]
[id="additional-resources_configuring-iam-ibm-cloud"]
== Additional resources

per updated guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

File: installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+For a detailed description of all available CCO credential modes and their supported platforms, see xref:../../authentication/managing_cloud_provider_credentials/about-cloud-credential-operator.adoc[About the Cloud Credential Operator].
+
+include::modules/cco-ccoctl-configuring.adoc[leveloffset=+1]
+include::modules/refreshing-service-ids-ibm-cloud.adoc[leveloffset=+1]
+//include::modules/manually-maintained-credentials-upgrade.adoc[leveloffset=+1]
+// Will need to revisit upgrade scenario for IBM Cloud; not needed until OCP 4.11. Tentative instructions have been added for reference later.
+
+[id="next-steps_configuring-iam-ibm-cloud"]
+== Next steps
```
Comment (ahardin-rh) [inline installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc:22]: [id="next-steps_configuring-iam-ibm-cloud"]
== Next steps

per updated guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

File: installing/installing_ibm_cloud_public/installing-ibm-cloud-customizations.adoc [hunk 3]
```diff
@@ -0,0 +1,61 @@
+[id="installing-ibm-cloud-customizations"]
+= Installing a cluster on IBM Cloud with customizations
+include::modules/common-attributes.adoc[]
+:context: installing-ibm-cloud-customizations
+
+toc::[]
+
+In {product-title} version {product-version}, you can install a customized cluster on infrastructure that the installation program provisions on IBM Cloud. To customize the installation, you modify parameters in the `install-config.yaml` file before you install the cluster.
+
+== Prerequisites
```
Comment (ahardin-rh) [inline installing/installing_ibm_cloud_public/installing-ibm-cloud-customizations.adoc:10]: [id="prerequisites_installing-ibm-cloud-customizations"]
== Prerequisites

File: installing/installing_ibm_cloud_public/preparing-to-install-on-ibm-cloud.adoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
+[id="preparing-to-install-on-ibm-cloud"]
+= Preparing to install on IBM Cloud
+include::modules/common-attributes.adoc[]
+:context: preparing-to-install-on-ibm-cloud
+
+toc::[]
+
+[id="{context}-prerequisites"]
```
Comment (ahardin-rh) [inline installing/installing_ibm_cloud_public/preparing-to-install-on-ibm-cloud.adoc:8]: [id="prerequisites_preparing-to-install-on-ibm-cloud"]
== Prerequisites

File: modules/installation-cis-ibm-cloud.adoc [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+* You have installed the link:https://www.ibm.com/cloud/cli[IBM Cloud CLI].
+
+.Procedure
+
+. If you do not already have an existing domain and registrar, you must acquire them. For more information, see IBM's link:https://cloud.ibm.com/docs/dns?topic=dns-getting-started[documentation].
+
+. Create a CIS instance to use with your cluster.
+
+.. Install the CIS plugin:
```
Comment (ahardin-rh) [inline modules/installation-cis-ibm-cloud.adoc:27]: plug-in

File: modules/installation-configuration-parameters.adoc [hunk 6]
```diff
@@ -1036,6 +1044,40 @@ The GCP Compute Engine System service account email, like `<service_account_name
...
+|====
+|Parameter|Description|Values
+
+|`platform.ibmcloud.resourceGroupName`
+|The name of an existing resource group to install your cluster to. This resource group must only be used for this specific cluster because the cluster components assume ownership of all of the resources in the resource group. If undefined, a new resource group is created for the cluster. ^1^
+|String, for example `existing_resource_group`.
+
+|`platform.ibmcloud.dedicatedHosts.profile`
+|The new dedicated host to create. If you specify a value for `platform.ibmcloud.dedicatedHosts.name`, this parameter is not required.
+|Valid IBM Cloud dedicated host profile, such as `cx2-host-152x304`. ^2^
```
Comment (ahardin-rh) [inline modules/installation-configuration-parameters.adoc:1064]: Looking at https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#alternative-footnote-styling-in-tables you might need brackets in the footnote callouts, but looks like it's rendering okay in the preview build. I would just double check this.

File: modules/installation-configuration-parameters.adoc [hunk 7]
```diff
@@ -1036,6 +1044,40 @@ The GCP Compute Engine System service account email, like `<service_account_name
...
+|String, for example `my-dedicated-host-name`.
+
+|`platform.ibmcloud.type`
+|The instance type for all IBM Cloud machines.
+|Valid IBM Cloud instance type, such as `bx2-8x32`. ^2^
+
+|====
+[.small]
+--
+1. Whether you define an existing resource group, or if the installer creates one, determines how the resource group is treated when the cluster is uninstalled. If you define a resource group, the installer removes all of the installer provisioned resources, but leaves the resource group alone; if a resource group is created as part of the installation, the installer removes all of the installer provisioned resources and the resource group.
```
Comment (ahardin-rh) [inline modules/installation-configuration-parameters.adoc:1077]: installer-provisioned resources

File: modules/installation-ibm-cloud-config-yaml.adoc [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+ifeval::["{context}" == "installing-ibm-cloud-network-customizations"]
+:with-networking:
+endif::[]
+ifeval::["{context}" != "installing-ibm-cloud-network-customizations"]
+:without-networking:
+endif::[]
+
+[id="installation-ibm-cloud-config-yaml_{context}"]
+= Sample customized `install-config.yaml` file for IBM Cloud
```
Comment (ahardin-rh) [inline modules/installation-ibm-cloud-config-yaml.adoc:14]: No markup in headings, so:
= Sample customized install-config.yaml file for IBM Cloud

File: modules/installation-ibm-cloud-config-yaml.adoc [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+====
+If you disable simultaneous multithreading, ensure that your capacity planning accounts for the dramatically decreased machine performance. Use larger machine types, such as `n1-standard-8`, for your machines if you disable simultaneous multithreading.
+====
+ifndef::openshift-origin[]
+<5> Whether to enable or disable FIPS mode. By default, FIPS mode is not enabled. If FIPS mode is enabled, the {op-system-first} machines that {product-title} runs on bypass the default Kubernetes cryptography suite and use the cryptography modules that are provided with {op-system} instead.
++
+[IMPORTANT]
+====
+The use of FIPS Validated / Modules in Process cryptographic libraries is only supported on {product-title} deployments on the `x86_64` architecture.
+====
```
Comment (ahardin-rh) [inline modules/installation-ibm-cloud-config-yaml.adoc:89]: can we replace the / with or?

File: modules/manually-maintained-credentials-upgrade.adoc [hunk 10]
```diff
@@ -61,6 +80,27 @@ It may take several minutes after adding the annotation for the upgradeable stat
...
 When the *Upgradeable* status in the *Conditions* section is *True*, you can begin the {product-title} upgrade.
+endif::ibm-cloud[]
+ifdef::ibm-cloud[]
++
+When the *Upgradeable* status in the *Conditions* section is *True*, you can begin the {product-title} upgrade.
+
+. Revoke the old service IDs and API Keys:
++
+[source,terminal]
++
```
Comment (ahardin-rh) [inline modules/manually-maintained-credentials-upgrade.adoc:94]: You don't need the line break between [source,terminal] and the start of the  code block

File: modules/installation-uninstall-clouds.adoc [hunk 11]
```diff
@@ -48,10 +48,49 @@ endif::gcp[]
 cluster.
 ifdef::ibm-cloud[]
 * You have configured the `ccoctl` binary.
+* You have installed the IBM Cloud CLI and installed or updated the VPC infrastructure service plug-in. For more information see "Prerequisites" in the link:https://cloud.ibm.com/docs/vpc?topic=vpc-infrastructure-cli-plugin-vpc-reference&interface=ui#cli-ref-prereqs[IBM VPC CLI documentation].
 endif::ibm-cloud[]
 
 .Procedure
+ifdef::ibm-cloud[]
```
Comment (mjpytlak) [inline modules/installation-uninstall-clouds.adoc:55]: @cjschaef Please take a pass at this procedure. In particular, can you define what `<infrastructure_id>` is? Thanks in advance.

File: modules/installation-uninstall-clouds.adoc [hunk 12]
```diff
@@ -48,10 +48,49 @@ endif::gcp[]
...
+
+.. Log in to the IBM Cloud using the CLI.
+.. To list the PVCs, run the following command:
++
+[source, terminal]
+----
+$ ibmcloud is volumes --resource-group-name <infrastructure_id>
+----
++
+For more information about listing volumes, see the link:https://cloud.ibm.com/docs/vpc?topic=vpc-infrastructure-cli-plugin-vpc-reference&interface=ui#volume-cli[IBM VPC CLI documentation].
```
Comment (jeffnowicki) [inline modules/installation-uninstall-clouds.adoc:71]: (minor) use "IBM Cloud VPC CLI"

File: modules/installation-uninstall-clouds.adoc [hunk 13]
```diff
@@ -48,10 +48,49 @@ endif::gcp[]
...
+
+.. To delete the PVCs, run the following command:
++
+[source, terminal]
+----
+$ ibmcloud is volume-delete --force <volume_id>
+----
++
+For more information about deleting volumes, see the link:https://cloud.ibm.com/docs/vpc?topic=vpc-infrastructure-cli-plugin-vpc-reference&interface=ui#volume-delete[IBM VPC CLI documentation].
```
Comment (jeffnowicki) [inline modules/installation-uninstall-clouds.adoc:80]: (minor) use "IBM Cloud VPC CLI"

**Review by jeffnowicki — APPROVED** [review]
I only have 2 minor comments.

File: installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc (hunk 1 above)
Comment (mjpytlak) [reply installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc:12]: Confirmed with Katherine earlier that .Additional resources is still OK, but the role and anchor ID is requiered.

Comment (mjpytlak) [reply installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc:12]: Done.

File: installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc (hunk 2 above)
Comment (mjpytlak) [reply installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc:22]: @ahardin-rh not sure if I need an update here. The heading already has an anchor ID that adheres to the standards.

File: installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc (hunk 1 above)
Comment (mjpytlak) [reply installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc:12]: Done.

Comment (mjpytlak) [reply installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc:12]: @ahardin-rh So based on the conversation in the peer review Slack channel, @kalexand-rh confirmed that using .`Additional resources` below a module is still legit.

File: installing/installing_ibm_cloud_public/installing-ibm-cloud-customizations.adoc (hunk 3 above)
Comment (mjpytlak) [reply installing/installing_ibm_cloud_public/installing-ibm-cloud-customizations.adoc:10]: Done.

File: modules/installation-cis-ibm-cloud.adoc (hunk 5 above)
Comment (mjpytlak) [reply modules/installation-cis-ibm-cloud.adoc:27]: Done.

File: modules/installation-ibm-cloud-config-yaml.adoc (hunk 8 above)
Comment (mjpytlak) [reply modules/installation-ibm-cloud-config-yaml.adoc:14]: Done.

File: modules/installation-ibm-cloud-config-yaml.adoc (hunk 9 above)
Comment (mjpytlak) [reply modules/installation-ibm-cloud-config-yaml.adoc:89]: Yep. Done.

File: modules/installation-configuration-parameters.adoc (hunk 6 above)
Comment (mjpytlak) [reply modules/installation-configuration-parameters.adoc:1064]: Updated to include `[]`

File: modules/installation-configuration-parameters.adoc (hunk 7 above)
Comment (mjpytlak) [reply modules/installation-configuration-parameters.adoc:1077]: Done.

File: modules/installation-uninstall-clouds.adoc (hunk 12 above)
Comment (mjpytlak) [reply modules/installation-uninstall-clouds.adoc:71]: Done.

File: modules/installation-uninstall-clouds.adoc (hunk 13 above)
Comment (mjpytlak) [reply modules/installation-uninstall-clouds.adoc:80]: Done.

File: modules/manually-maintained-credentials-upgrade.adoc (hunk 10 above)
Comment (mjpytlak) [reply modules/manually-maintained-credentials-upgrade.adoc:94]: Done.

**Comment by mjpytlak** [comment]
@ahardin-rh Peer review comments addressed. Thank you.
@jeffnowicki Addressed your comments. Thank you.

@pamoedom Outside of your concern about the TOC, do you have additional feedback on the last round of edits? If not, please approve so I can merge this doc by COB 2/25. Thank you.

**Comment by ahardin-rh** [comment]
All latest updates LGTM!

**Pull request merged**